src/mme-app/contextManager/dataBlocks.cpp

```diff
@@ -628,20 +709,64 @@ namespace mme
 
 	
 	/******************************************************************************
-	* sets BearerContext
+	* add BearerContext
+	******************************************************************************/
+	bool SessionContext::addBearerContext(BearerContext* BearerContext_i)
+	{
+	        if( BearerContext_m.size() <= 15 )
+    		        BearerContext_m.push_back(BearerContext_i);
+    		else
+    		        return false;
+    		return true;
+	}
+	
+	/******************************************************************************
+	* remove BearerContext
+	******************************************************************************/
+	void SessionContext::removeBearerContext(BearerContext* BearerContext_i)
+	{
+    	    	BearerContext_m.remove(BearerContext_i);
+	}
+		
+	/******************************************************************************
+	* returns BearerContextList
+	******************************************************************************/
+	std::list<BearerContext*>& SessionContext::getBearerContextContainer()
+	{
+		return BearerContext_m;
+	}
+	
+	/******************************************************************************
+	* find BearerContext ByBearerId
 	******************************************************************************/
-	void SessionContext::setBearerContext( BearerContext* BearerContextp )
+	BearerContext* SessionContext::findBearerContextByBearerId( uint8_t bearerId_i)
 	{
-		BearerContext_mp = BearerContextp;
+		for( auto &it : BearerContext_m)
+    		{
+        		if(it->getBearerId() == bearerId_i)
+        		{
+            			return it;
+        		}
+    		}
+    		return NULL;
 	}
 	
 	/******************************************************************************
-	* returns BearerContext
+	* find BearerContext ByS1uSgwUserFteid
 	******************************************************************************/
-	BearerContext* SessionContext::getBearerContext()
+	BearerContext* SessionContext::findBearerContextByS1uSgwUserFteid( const Fteid& s1uSgwUserFteid_i)
```

Comment:
What is the use case for this lookup table ?

Reply:
We *may* need this when we send create bearer response..bearers within the request are correlated using the s1sgw fteids.